README.md

```diff
@@ -118,3 +101,171 @@ Remote debugging using :2000
+Memory region         Used Size  Region Size  %age Used
+             rom:      135160 B       128 KB    103.12%
+             ram:        5708 B        20 KB     27.87%
+collect2: error: ld returned 1 exit status
+```
+
+This is not unexpected, as some hardware probe have limited space for firmware, you can get around it by disabling some features or target support:
+``` sh
+meson setup --reconfigure -Dtargets=cortexm,stm -Drtt_support=false
+```
+
+#### Building Black Magic Debug App
+
+The Black Magic Debug App (BMDA) is always build by default, even for firmware builds, so long as all it's dependencies are found, you can find the executable under the build directory, named simply `blackmagic`
+
+If you wish to build only the BMDA, you can set the hardware `probe` option to an empty string `-Dprobe=''`, this is the default value:
+``` sh
+meson setup build
+```
+
+You now should have a `build` directory from where you can build the app, this is also where your executable will appear.
+
+The command `meson compile` will build the default targets:
+```sh
+cd build
+meson compile
+```
+
+Ypu should now see the resulting executable in `build`:
+```
+blackmagic
+```
+
+Most hosted building is done on and for Linux. BMP-hosted for windows can also be build with Mingw on Linux.
+
+### Changing the build configuration
+
+You may change the configuration at any time after configuring the project, if may also change the default options while configuring for the first time.
+
+Changing options at configure time:
+``` sh
+meson setup build -Ddebug_output=true
+```
+
+Changin options after configuration:
+``` sh
+meson setup --reconfigure -Dtargets=cortexm,stm
+```
+
+Keep in mind that when changing the configured probe, the other default options will not change, and you may end up with a configuration that does not make sense for it, so in such cases it's best to use the *cross-file* for the probe, not just change the `probe` option.
+``` sh
+meson setup --reconfigure --cross-file=../cross-file/bluepill.ini
+```
+
+You can have multiple build directories! So if you are regularly building firmware for multiple probes we would recomend keeping an individual build directory configured for each one.
+
+### Working with an existing clone (used before the new meson buildsystem was introduced)
+
+If you are working with an existing clone of the project where you used the old `make` buildsystem, or simply initialized the submodules, you may encounter issues while trying the new `meson` workflow, this may look something like:
+```
+fatal: Fetched in submodule path 'deps/libopencm3', but it did not contain bed4a785eecb6c9e77e7491e196565feb96c617b. Direct fetching of that commit failed.
+```
+
+To get around this try running the following commands:
+``` sh
+git submodule deinit --force --all
+meson subprojects purge --confirm
+```
+
+If this does not resolve the issue, consider making a fresh clone of the repository.
```

Comment:
Doing `git remote set-url` in the submodule is guaranteed to work, we'd prefer not to encourage this mentality.